src/iface/ethernet.rs

```diff
@@ -233,16 +323,30 @@ impl<'a, 'b, 'c, DeviceT: Device + 'a> Interface<'a, 'b, 'c, DeviceT> {
 
         Ok(())
     }
+    
+    fn keep_ethernet_frame<'frame, T: AsRef<[u8]>>(&self, eth_frame: &EthernetFrame<&T>) -> bool {
+      // the multicast frame belongs to one of the subscribed groups
+      if self.eth_mcast_addr.contains(&eth_frame.dst_addr()) {
```

Comment:
Maybe check if `eth_frame.dst_addr()` is multicast before trying to do a hash lookup?